plugin/auth_ldap/src/plugin_simple.cc

```diff
@@ -0,0 +1,189 @@
+/* Copyright (c) 2019 Francisco Miguel Biete Banon. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   51 Franklin Street, Suite 500, Boston, MA 02110-1335 USA */
+
+#include "plugin/auth_ldap/include/plugin_simple.h"
+#include "plugin/auth_ldap/include/connection.h"
+#include "plugin/auth_ldap/include/plugin_common.h"
+#include "plugin/auth_ldap/include/plugin_log.h"
+#include "plugin/auth_ldap/include/plugin_variables.h"
+#include "plugin/auth_ldap/include/pool.h"
+
+#include <condition_variable>
+#include <mutex>
+
+#define PASSWORD_QUESTION "\5"
+
+mysql::plugin::auth_ldap::Ldap_logger *g_logger_server;
+
+MYSQL_PLUGIN auth_ldap_simple_plugin_info;
+
+namespace {
+std::mutex active_m;
+std::condition_variable active_cv;
+int active_connections = 0;
+}  // namespace
+
+// Declaration to access the name of the SYS_VAR
+struct SYS_VAR {
+  MYSQL_PLUGIN_VAR_HEADER;
+};
+
+template <typename Copy_type>
+void update_sysvar(THD *, SYS_VAR *var, void *var_ptr, const void *value) {
+  // Update the value
+  *(Copy_type *)var_ptr = *(Copy_type *)const_cast<void *>(value);
+
+  if (strcmp(var->name, "authentication_ldap_simple_log_status") == 0)
+    g_logger_server->set_log_level(
+        static_cast<mysql::plugin::auth_ldap::ldap_log_level>(log_status));
+  else {
+    connPool->reconfigure(init_pool_size, max_pool_size,
+                          STR_OR_EMPTY(server_host), server_port, ssl, tls,
+                          STR_OR_EMPTY(ca_path), STR_OR_EMPTY(bind_root_dn),
+                          STR_OR_EMPTY(bind_root_pwd));
+    connPool->debug_info();
+  }
+}
+
+static int auth_ldap_simple_init(MYSQL_PLUGIN plugin_info) {
+  g_logger_server = new mysql::plugin::auth_ldap::Ldap_logger();
```

Comment:
Why do we need custom logger at all? Can existing MySQL logging infrastructure be reused here?

Reply:
I also asked this, but looks like the enterprise plugin also has this feature.